adobecom/milo
# MWPW-143143: Improvements for Upgrade modal

### Links
Resolves https://jira.corp.adobe.com/browse/MWPW-143143 
Tacocat PR: https://git.corp.adobe.com/wcms/tacocat.js/pull/592

### Summary
* pass promo code, set on Upgrade Offer CTA to the Upgrade Modal
* don't show the Upgrade modal, unless 'upgrade=true' flag is present on Merch Link
* Upgrade modal to not depend on 'entitlement' flag (to enable the case when GWP needs a merch link to have Upgrade feature but not show the TWP/D2P modals).

### How To Test

- login on https://www.stage.adobe.com/ with an eligible test user
- open [test page ](https://main--cc--adobecom.hlx.page/cc-shared/fragments/drafts/commerce/testpage?milolibs=upromo)
- verify CTAs with 'upgrade=true' flag show modal
- re-login with not eligible test user and verify all links lead to Checkout Page

![image](https://github.com/adobecom/milo/assets/6617234/63828972-6984-485d-a0f4-1d17496f234c)


**Test URLs:**
- Before: https://main--milo--adobecom.hlx.live/drafts/mariia/pr/price-cta?georouting=off&martech=off
- After: https://upromo--milo--adobecom.hlx.live/drafts/mariia/pr/price-cta?georouting=off&martech=off
